modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -1702,264 +1719,549 @@ def tree_reduce(
+            If the argument is a scalar type, then that type is assigned to each result column.
 
         Returns
         -------
         PandasDataframe
-            A new dataframe.
-
-        Notes
-        -----
-        The data shape is not changed (length and width of the table).
+            New Modin dataframe.
         """
-        new_partitions = self._partition_mgr_cls.map_axis_partitions(
-            axis, self._partitions, func, keep_partitioning=True
-        )
+        new_partitions = self._partition_mgr_cls.map_partitions(self._partitions, func)
         return self.__constructor__(
             new_partitions,
             self._index_cache,
             self._columns_cache,
             self._row_lengths_cache,
             self._column_widths_cache,
+            dtypes=dtypes,
         )
 
-    def infer_objects(self) -> "PandasDataframe":
+    @lazy_metadata_decorator(apply_axis="both")
+    def _map_axis(
+        self,
+        func: Callable,
+        axis: AxisInt,
+        *,
+        other,
+        join_type,
+        labels,
+        dtypes,
+        copy_dtypes,
+    ) -> "PandasDataframe":
         """
-        Attempt to infer better dtypes for object columns.
+        Perform a row or column-wise map across the dataframe.
 
-        Attempts soft conversion of object-dtyped columns, leaving non-object and unconvertible
-        columns unchanged. The inference rules are the same as during normal Series/DataFrame
-        construction.
+        Parameters
+        ----------
+        func : callable
+            Function to apply.
+        axis : {0, 1}
+            Axis to broadcast over.
+        other : PandasDataframe
+            Modin DataFrame to broadcast.
+        join_type : str, default: "left"
+            Type of join to apply.
+        labels : {"keep", "replace", "drop"}
+            Whether keep labels from `self` Modin DataFrame, replace them with labels
+            from joined DataFrame or drop altogether to make them be computed lazily later.
+        dtypes : pandas.Series, optional
+            Data types of the result. This is an optimization
+            because there are functions that always result in a particular data
+            type, and allows us to avoid (re)computing it.
+        copy_dtypes : bool
+            If True, the dtypes of the resulting dataframe are copied from the original,
+            and the ``dtypes`` argument is ignored.
 
         Returns
         -------
         PandasDataframe
-            A new PandasDataframe with the inferred schema.
+            New Modin DataFrame.
         """
-        obj_cols = [
-            col for col, dtype in enumerate(self.dtypes) if is_object_dtype(dtype)
-        ]
-        return self.infer_types(obj_cols)
+        if copy_dtypes:
+            dtypes = self._dtypes
+        if other is None:
+            new_partitions = self._partition_mgr_cls.map_partitions(
+                self._partitions, func
+            )
```

Comment:
why do we ignore `axis` here? why does the `.map_partitions` call is inside of `_map_axis` that's supposed to call function axis-wise only?